### PR TITLE
SB-509: npm package generator + `npm publish` UnitTest [ready]

### PR DESCRIPTION
**Pull request by sbespalov**

### CODE REVIEW

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NpmPackageGenerator.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    {
+        return new NpmPackageGenerator();
+    }
+
+    private NpmPackageGenerator()
+    {
+        super();
+    }
+
+    public NpmPackageGenerator of(NpmArtifactCoordinates c)
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NpmPackageGenerator.java]: What is the purpose of this constructor and why is the name `of`?

**Comment by sbespalov** [comment]
It's not a constructor, this aproach called Builder pattern :) methods
return ‘this‘ so you can make a chined calls to construct instance you need

28 нояб. 2017 г. 23:25 пользователь "Martin Todorov" <
notifications@github.com> написал:

**Comment by sbespalov** [comment]
So you construct npm package OF coordinates IN some base  folder :)

29 нояб. 2017 г. 7:10 пользователь "Serge Bespalov" <
serge.bespalov@gmail.com> написал:

**Comment by carlspring** [comment]
As always, thank you explaining your reasoning and for the lecture in
programming! :)

**Pull request merged**